src/cljs/rems/status_modal.cljs

```diff
@@ -47,6 +47,55 @@
        (when @state [component @state])
        [:button.btn.btn-secondary {:on-click #(reset! state (assoc opened-state :on-close on-close))} "Open modal"]])))
 
+(defn status-modal-opts
+  "Returns a map of modal options, that can be used by status-modal/situational-status-modal.
+
+   The returned value is a map containing various event handlers, and most importantly,
+   a state atom under the key :state-atom."
+  [{:keys [on-success on-pending on-error on-close-after-success on-close-after-error] :as modal-options}]
+  (let [state (r/atom nil)]
+    (merge
+     modal-options
+     {:state-atom state
+      :on-pending #(do
+                     (swap! state assoc :status :pending)
+                     (when on-pending (on-pending)))
+      :on-success #(do
+                     (swap! state assoc :status :saved)
+                     (when on-success (on-success)))
+      :on-error #(do
+                   (swap! state assoc :status :failed :error %)
+                   (when on-error (on-error)))
+      :on-close-after-success #(do
+                                 (reset! state nil)
+                                 (when on-close-after-success (on-close-after-success)))
+      :on-close-after-error #(do
+                               (reset! state nil)
+                               (when on-close-after-error (on-close-after-error)))})))
+
+(defn situational-status-modal
```

Comment:
Why this name? The file name is `status-modal` so this could be `component` if there are no other components here or just `status-modal`?

Reply:
situational-status-modal is now removed